src/ir/context.rs

```diff
@@ -865,6 +865,27 @@ If you encounter an error missing from this list, please file an issue or a PR!"
                 "struct" | "super" | "trait" | "true" | "type" | "typeof" |
                 "unsafe" | "unsized" | "use" | "virtual" | "where" |
                 "while" | "yield" | "bool" | "_" => true,
+
+                "f32" |
+                "f64" |
+                "usize" |
+                "isize" |
+                "u128" |
+                "i128" |
+                "u64" |
+                "i64" |
+                "u32" |
+                "i32" |
+                "u16" |
+                "i16" |
+                "u8" |
+                "i8" => {
+                    if !self.options.allow_primitive_collisions {
```

Comment:
I don't think this option is particularly useful if its only side-effect is `error!`-ing out. Having an option that preserves the current behavior may be worth it, but probably not.

So I think it's ok to do this minor breaking change. If somebody had a use-case for the pre-existing behavior we can add such an option.

Reply:
Well yet you are referring to the previous implementation by @jhod0 that I rebased as a starting point. I then changed it to warn! by default and mangle when the flag is set.